EIPS/draft_unlimited_swapdup.md

```diff
@@ -0,0 +1,57 @@
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Unlimited SWAP and DUP instructions
+    Author: Alex Beregszaszi
+    Type: Standard Track
+    Category: Core
+    Status: Draft
+    Created: 2017-07-03
+
+
+## Abstract
+
+`SWAP` and `DUP` instructions are limited to a stack depth of 16. Introduce two new instructions, `SWAPn` and `DUPn`, which lift this limitation and allow accessing the stack up to its full depth of 1024 items.
+
+## Motivation
+
+Implementing higher level constructs, such as functions, on top of EVM will result in a list of input and output parameters as well as an instruction offset to return to.
+
+The number of these arguments (or stack items) can easily exceed 16 and thus will require extra care from a compiler to lay them out in a way that all of them are still accessible.
+
+Introducing `SWAPn` and `DUPn` will provide an option to compilers to simplify accessing deep stack items at the price of possibly increased gas costs.
+
+## Specification
+
+Instructions `DUPn` (`0xb0`) and `SWAPn` (`0xb1`) are introduced, which take the top item from stack (referred to as `n`).
+
+If `n` exceeds 1024 or the current stack depth is less than `n`, then a stack underflow exception is issued. If the current stack depth is at the limit, a stack overflow exception is issued.
```

Comment:
When a stack underflow exception is issued, what should happen to the contract execution?  Would it be equivalent to `REVERT` with a certain output?

Does it ever happen that the stack depth is still at the limit after popping `n`?